src/shape/trimesh.rs

```diff
@@ -172,18 +172,25 @@ pub struct TriMeshTopology {
     pub half_edges: Vec<TopoHalfEdge>,
 }
 
-impl TriMeshTopology {
-    #[cfg(feature = "dim3")]
-    pub(crate) fn face_half_edges_ids(&self, fid: u32) -> [u32; 3] {
-        let first_half_edge = self.faces[fid as usize].half_edge;
-
-        let mut result = [first_half_edge; 3];
-        for k in 1..3 {
-            let half_edge = self.half_edges[result[k - 1] as usize];
-            result[k] = half_edge.next;
+#[cfg(all(feature = "std", feature = "cuda"))]
+impl TriMeshTopology<CudaStorage> {
+    fn as_device_ptr(&self) -> TriMeshTopology<CudaStoragePtr> {
+        TriMeshTopology {
+            vertices: self.vertices.as_device_ptr(),
+            faces: self.faces.as_device_ptr(),
+            half_edges: self.half_edges.as_device_ptr(),
         }
+    }
+}
 
-        result
+#[cfg(all(feature = "std", feature = "cuda"))]
+impl TriMeshTopology<DefaultStorage> {
+    fn to_cuda(&self) -> CudaResult<TriMeshTopology<CudaStorage>> {
+        Ok(TriMeshTopology {
+            vertices: CudaArray1::new(&self.vertices)?,
+            faces: CudaArray1::new(&self.faces)?,
+            half_edges: CudaArray1::new(&self.half_edges)?,
+        })
```

Comment:
These should be removed. Looks like another merge artifact (there is no cuda support anymore).